Content.Shared/Hallucinations/HallucinationsComponent.cs

```diff
@@ -0,0 +1,74 @@
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom;
+
+namespace Content.Shared.Hallucinations;
+
+[RegisterComponent]
+public sealed partial class HallucinationsComponent : Component
+{
+    [DataField(customTypeSerializer: typeof(TimeOffsetSerializer)), ViewVariables(VVAccess.ReadWrite)]
+    public TimeSpan NextSecond = TimeSpan.Zero;
+
+    /// <summary>
+    /// How far from humanoid can appear hallucination
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadWrite)]
+    public float Range = 7f;
+
+    /// <summary>
+    /// How often (in seconds) hallucinations spawned
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadWrite)]
```

Comment:
This needs to be networked, TimeOffsetSerializer doesn't handle pausing on its own but we have a sourcegen version for that now, and datafield + VVRW is pointless.